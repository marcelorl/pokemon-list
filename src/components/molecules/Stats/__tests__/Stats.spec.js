import createTestContext from 'react-cosmos-test/enzyme'

import fixture from '../Stats.fixture'

const { mount, getWrapper } = createTestContext({ fixture })

beforeEach(mount)

describe('#Stats', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('should have 3 progress bars', () => {
    expect(getWrapper().find('Progress')).toHaveLength(3)
  })
})
