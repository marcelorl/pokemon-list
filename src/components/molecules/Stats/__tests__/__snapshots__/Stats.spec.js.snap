// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Stats renders component 1`] = `
<Stats
  items={
    Array [
      Object {
        "name": "test1",
        "value": 10,
      },
      Object {
        "name": "test2",
        "value": 20,
      },
      Object {
        "name": "test3",
        "value": 30,
      },
    ]
  }
>
  <div
    key="0"
  >
    <div
      className="text-center"
    >
      test1
    </div>
    <Progress
      max={100}
      tag="div"
      value={10}
    >
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin="0"
          aria-valuenow={10}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "10%",
            }
          }
        />
      </div>
    </Progress>
  </div>
  <div
    key="1"
  >
    <div
      className="text-center"
    >
      test2
    </div>
    <Progress
      max={100}
      tag="div"
      value={20}
    >
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin="0"
          aria-valuenow={20}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "20%",
            }
          }
        />
      </div>
    </Progress>
  </div>
  <div
    key="2"
  >
    <div
      className="text-center"
    >
      test3
    </div>
    <Progress
      max={100}
      tag="div"
      value={30}
    >
      <div
        className="progress"
      >
        <div
          aria-valuemax={100}
          aria-valuemin="0"
          aria-valuenow={30}
          className="progress-bar"
          role="progressbar"
          style={
            Object {
              "width": "30%",
            }
          }
        />
      </div>
    </Progress>
  </div>
</Stats>
`;
